#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

From c8cdb4c9a9bb590b8785b483cf71657e20e0b92d Mon Sep 17 00:00:00 2001
From: Andre Heinecke <aheinecke@intevation.de>
Date: Thu, 11 Aug 2016 15:59:57 +0200
Subject: [PATCH] Fix loop logic error in new server name detection

* src/engine-assuan.c (get_uiserver_name): Fix name loop.
---
 src/engine-assuan.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/engine-assuan.c b/src/engine-assuan.c
index 32dbae8..c60713b 100644
--- a/src/engine-assuan.c
+++ b/src/engine-assuan.c
@@ -353,7 +353,7 @@ get_uiserver_name (void)
                                  "gpa.exe",
                                  "bin\\gpa.exe",
                                  NULL};
-  const char *tmp = NULL;
+  const char **tmp = NULL;

   dir = get_gpg4win_dir ();
   if (!dir)
@@ -384,14 +384,14 @@ get_uiserver_name (void)
       return name;
     }
   /* Fallbacks */
-  for (tmp = *server_names; *tmp; tmp++)
+  for (tmp = server_names; *tmp; tmp++)
     {
       if (name)
         {
           xfree (name);
         }
-      name = xmalloc (strlen (dir) + strlen (tmp) + extra_arglen + 2);
-      strcpy (stpcpy (stpcpy (name, dir), "\\"), tmp);
+      name = xmalloc (strlen (dir) + strlen (*tmp) + extra_arglen + 2);
+      strcpy (stpcpy (stpcpy (name, dir), "\\"), *tmp);
       for (p = name; *p; p++)
         if (*p == '/')
           *p = '\\';
--
2.1.4
